libraries/aws/shadow/include/shadow.h

```diff
@@ -0,0 +1,546 @@
+     * without _DOCUMENTS suffix. This rule is suppressed for naming consistency with other Shadow
+     * header field and value string and length macros in this file.*/
+    /* coverity[misra_c_2012_rule_5_2_violation] */
+    /* coverity[other_declaration] */
+    SHADOW_TOPIC_STRING_TYPE_UPDATE_DOCUMENTS,
+
+    /* MISRA Rule 5.2 flags the following member's name as ambiguous from the one
+     * without _DELTA suffix. This rule is suppressed for naming consistency with other Shadow
+     * header field and value string and length macros in this file.*/
+    /* coverity[misra_c_2012_rule_5_2_violation] */
+    /* coverity[other_declaration] */
+    SHADOW_TOPIC_STRING_TYPE_UPDATE_DELTA,
+    SHADOW_TOPIC_STRING_TYPE_MAX_NUM
+} ShadowTopicStringType_t;
+
+/**
+ * @brief Return codes from Shadow functions.
+ */
+typedef enum ShadowStatus
+{
+    SHADOW_STATUS_SUCCESS = 0,                          /**< @brief Shadow function success. */
+    SHADOW_STATUS_FAIL,                                 /**< @brief Shadow function encountered error. */
+    SHADOW_STATUS_BAD_PARAMETER,                        /**< @brief Input parameter is invalid. */
+    SHADOW_STATUS_BUFFER_TOO_SMALL,                     /**< @brief The provided buffer is too small. */
+    SHADOW_STATUS_THINGNAME_PARSE_FAILED,               /**< @brief Could not parse the thing name. */
+    SHADOW_STATUS_SHADOW_MESSAGE_TYPE_PARSE_FAILED      /**< @brief Could not parse the shadow type. */
+} ShadowStatus_t;
+
+/*------------------------ Shadow library functions -------------------------*/
+
+/**
+ * @brief The common prefix of all Shadow MQTT topics
+ *        from here https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html.
+ */
+#define SHADOW_TOPIC_PREFIX_STRING                                  "$aws/things/"
+
+/**
+ * @brief The length of #SHADOW_TOPIC_PREFIX.
+ */
+#define SHADOW_TOPIC_PREFIX_LENGTH                                  ( ( uint16_t ) ( sizeof( SHADOW_TOPIC_PREFIX_STRING ) - 1U ) )
+
+/**
+ * @brief The string representing a Shadow "DELETE" operation in a Shadow MQTT topic.
+ */
+
+/* MISRA Rule 5.4 flags the following macro's name as ambiguous from the
+ * one without suffix. This rule is suppressed for naming consistency with
+ * other Shadow header field and value string and length macros in this file.*/
+/* coverity[other_declaration] */
+#define SHADOW_TOPIC_OPERATION_STRING_DELETE                        "/shadow/delete"
+
+/**
+ * @brief The length of #SHADOW_TOPIC_OPERATION_STRING_DELETE.
+ */
+
+/* MISRA Rule 5.4 flags the following macro's name as ambiguous from the
+ * one without suffix. This rule is suppressed for naming consistency with
+ * other Shadow header field and value string and length macros in this file.*/
+/* coverity[other_declaration] */
+#define SHADOW_TOPIC_OPERATION_LENGTH_DELETE                        ( ( uint16_t ) ( sizeof( SHADOW_TOPIC_OPERATION_STRING_DELETE ) - 1U ) )
```

Comment:
In the sign-off review, we get suggestions to avoid abbreviation
LEN --> Length,
STR --> String,
I'm not sure if we should change it to OP but if we decide we can't get a pass for HTTP and Shadow.
We can try to make this change

Reply:
We don't necessarily have to use abbreviations, we can just reorganize it so the first 30 or so characters are unique. What about something like `SHADOW_DELETE_SUBSTRING_LENGTH`? Just a possible suggestion, since it might be confused with the macros you use below, so if you think there's a better identifier, then please use it.

Reply:
decided to take your suggestion with OPERATION --> OP (Thank you):

fixed it in commit [8da68c5](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1081/commits/8da68c5f3ccc011a9ab8f58349cf6273c42ec19c)

shorten the macros and changed XX_OPERATION_XX to XX_OP_XX to avoid misra 5.4, comments for the macro explains that the macro is used for operations


resolving this comment

feel free to let me know if it needs further changes, thank you